include/oneapi/dpl/pstl/hetero/numeric_impl_hetero.h

```diff
@@ -108,19 +121,210 @@ __pattern_transform_reduce(_ExecutionPolicy&& __exec, _ForwardIterator __first,
+                }
+            });
+        });
+        __event.wait();
+    }
+
+    template<::uint16_t _ElemsPerItem, ::uint16_t _WGSize, bool _IsFullGroup, typename _Policy, typename _InRng, typename _OutRng, typename _InitType, typename _BinaryOperation, typename _UnaryOp>
+    static void __launch_static_bounds_scan(_Policy const & __policy, _InRng __in, _OutRng __out, std::size_t __n, _InitType __init, const _BinaryOperation& __bin_op, const _UnaryOp& __unary_op)
+    {
+        using _RangeValueType = oneapi::dpl::__internal::__value_t<_InRng>;
+        using _ValueType = decltype(__bin_op(std::declval<_RangeValueType>(), std::declval<_RangeValueType>()));
+        using _CustomName = typename _Policy::kernel_name;
+        using _GroupScanKernel =
+                     oneapi::dpl::__par_backend_hetero::__internal::__kernel_name_generator<__par_backend_hetero::__scan_single_wg_kernel, _CustomName, _BinaryOperation, std::integral_constant<bool, _Inclusive>, std::integral_constant<bool, _IsFullGroup>, _InRng, _OutRng, std::integral_constant<::std::size_t, _ElemsPerItem>, std::integral_constant<std::size_t, _WGSize>>;
+
+
+        constexpr ::uint32_t __elems_per_wg = _ElemsPerItem * _WGSize;
+
+        auto __event = __policy.queue().submit([&](sycl::handler& __hdl) {
+            oneapi::dpl::__ranges::__require_access(__hdl, __in, __out);
+
+            auto __lacc = __dpl_sycl::__local_accessor<_ValueType>(sycl::range<1>{__elems_per_wg}, __hdl);
+
+            __hdl.parallel_for<__par_backend_hetero::__tunable_kernel_name<_GroupScanKernel, __elems_per_wg, _IsFullGroup, _Inclusive>>(sycl::nd_range<1>(_WGSize, _WGSize), [=](sycl::nd_item<1> __self_item) {
+                const auto& __group = __self_item.get_group();
+                const auto& __subgroup = __self_item.get_sub_group();
+                const auto __item_id = __self_item.get_local_linear_id();
+                int __subgroup_id = __subgroup.get_group_id();
+                int __id_in_subgroup = __subgroup.get_local_id();
+                int __subgroup_size = __subgroup.get_local_linear_range();
+
+                if constexpr (_IsFullGroup)
+                {
+                    #pragma unroll
+                    for (::uint16_t __i = 0; __i < _ElemsPerItem; ++__i)
+                    {
+                       auto __idx = __i * _WGSize + __subgroup_id * __subgroup_size;
+                       auto __val = __unary_op(__subgroup.load(__in.begin() + __idx));
+                       __subgroup.store(__lacc.get_pointer() + __idx, __val);
+                    }
+                }
+                else
+                {
+                    #pragma unroll
+                    for (::uint16_t __i = 0; __i < _ElemsPerItem; ++__i)
+                    {
+                       auto __idx = __i * _WGSize + __item_id;
+                       auto __val = __unary_op(__idx < __n ? __in[__idx] : unseq_backend::__known_identity<_BinaryOperation, _ValueType>);
```

Comment:
Applying the `__unary_op()` to the `__known_identity` makes it potentially no longer the identity, because it is identity is for `__bin_op` and `_ValueType`.  However, I don't believe it matters what we fill here as long as it is a valid input, all values which are impacted by these will be discarded before being returned to global memory.  

If it is a performance improvement to only apply the `__unary_op` to valid inputs, then we may want to change this.

Reply:
I agree that it is benign to apply the unary operator to the known identity since those values will be discarded anyway.

For the performance comment, I believe the common case would where the unary operator is a no-op and so it will not be an issue in that case. However, if we find that there are instances where the unary operator is a heavy function, we can consider eliding the calls to the unary op for the out-of-bounds elements.

Reply:
As the filling of identity (or any values) are removed in the most recent commit, I think we can call this resolved.  

Also, I'll note despite the lack of using the identity, I do think we need to continue checking for a `has_known_identity` because this allows us to know that we are using one of the class of binary operators that `sycl::joint_*_scan` supports.  I'm not aware of a better way to make that check (though one may exist).